server/src/main/java/com/cloud/event/Cadf.java

```diff
@@ -0,0 +1,286 @@
+
+            //eventtarget is event's Target
+            //eventaction is event's Action
+        }
+
+        //sets EventType, Action and Target
+        //Answers to WHAT, ONWHAT, TOWHERE
+        setCADFAction(eventaction);
+
+        eventType = getCADFEventType();
+
+
+        //Answers to ONWHAT and TOWHERE
+        target = new Resource(getCADFResourceName(eventtarget), getCADFResourceUUID(eventtarget));
+
+
+        //sets Outcome
+        //Answers to WHAT
+        mapEventStateToCADFTaxonomy(event.getState().toString());
+
+        //must add reason
+
+        //Answers to WHEN
+        //_eventTime must be in UTC format
+        if (event.getCreateDate() != null) {
+            eventTime = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.Z").format(event.getCreateDate());
+        } else {
+            eventTime="";
+        }
+
+        //Answers to WHO and FROMWHERE
+        //"USER" CS Resource corresponds to Taxonomies.Resource.DATA_SECURITY_ACCOUNT_USER CADF Resource
+
+        initiator = new Resource(Taxonomies.Resource.DATA_SECURITY_ACCOUNT_USER,
+                getCADFResourceUUID("USER"));
+
+
+        //Answers to WHERE
+
+        observer = new Resource(Taxonomies.Resource.DATA_SECURITY,
+                getCADFResourceUUID("SYSTEM.MONITOR"));
+
+        try {
+            InetAddress inetAddress = InetAddress.getLocalHost();
+            observer.host = new Resource.Host();
+            observer.host = new Resource.Host(UUID.nameUUIDFromBytes(inetAddress.getHostName().getBytes()).toString(),
+                    inetAddress.getHostAddress(), null , null);
+
+        } catch (UnknownHostException e) {
+            s_logger.error(e.getMessage());
+        }
+        if (eventExtraInformation != null &&
+                eventExtraInformation.get("initiator_host") != null &&
+                eventExtraInformation.get("initiator_user-agent") != null &&
+                !eventExtraInformation.get("initiator_host").isEmpty()) {
+            initiator.host = new Resource.Host(UUID.nameUUIDFromBytes(eventExtraInformation.get("initiator_host").getBytes()).toString(),
+                    eventExtraInformation.get("initiator_host"),
+                    eventExtraInformation.get("initiator_user-agent"),
+                    null);
+        }
+        if (eventExtraInformation != null &&
+                eventExtraInformation.get("initiator_userid") != null &&
+                !eventExtraInformation.get("initiator_userid").isEmpty()) {
+            initiator.userid = eventExtraInformation.get("initiator_userid");
+        }
+        if (eventExtraInformation != null &&
+                eventExtraInformation.get("initiator_csAccountName") != null &&
+                !eventExtraInformation.get("initiator_csAccountName").isEmpty()) {
+            initiator.csAccountName = eventExtraInformation.get("initiator_csAccountName");
+        }
+    }
```

Comment:
this constructor is large. any chance you could factor out for instance, initInitiator(). to improve readability?